pyramid/scripts/pserve.py

```diff
@@ -892,25 +893,34 @@ def check_reload(self):
                 pyc = True
             else:
                 pyc = False
-            if filename not in self.module_mtimes:
-                self.module_mtimes[filename] = mtime
-            elif self.module_mtimes[filename] < mtime:
+            old_mtime = self.module_mtimes.get(filename)
+            self.module_mtimes[filename] = mtime
+            if old_mtime is not None and old_mtime < mtime:
+                self.pending_reload = True
                 if pyc:
                     filename = filename[:-1]
-                # check if a file has syntax errors. If so, track it until it's fixed
-                for filename_promise in self.syntax_error_files + [filename]:
-                    try:
-                        py_compile.compile(filename_promise, doraise=True)
-                    except py_compile.PyCompileError:
-                        print("%s has a SyntaxError; NOT reloading." % filename_promise)
-                        if filename_promise not in self.syntax_error_files:
-                            self.syntax_error_files.append(filename_promise)
-                    else:
-                        if filename_promise in self.syntax_error_files:
-                            self.syntax_error_files.remove(filename_promise)
-                if not self.syntax_error_files:
-                    print("%s changed; reloading..." % filename)
-                    return False
+                is_valid = True
+                if filename.endswith('.py'):
+                    is_valid = self.check_syntax(filename)
+                if is_valid:
+                    print("%s changed ..." % filename)
+        if self.pending_reload and not self.syntax_error_files:
+            self.pending_reload = False
+            return False
+        return True
+
+    def check_syntax(self, filename):
+        # check if a file has syntax errors.
+        # If so, track it until it's fixed.
+        try:
+            py_compile.compile(filename, doraise=True)
+        except py_compile.PyCompileError:
+            print("%s has a SyntaxError; NOT reloading." % filename)
```

Comment:
But the information you're looking for is what files to fix to get reloader to work again